src/Sylius/Bundle/AdminBundle/Controller/Dashboard/StatisticsController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Controller\Dashboard;
+
+use Sylius\Component\Core\Dashboard\DashboardStatisticsProviderInterface;
+use Sylius\Component\Core\Dashboard\Interval;
+use Sylius\Component\Core\Dashboard\SalesDataProviderInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+final class StatisticsController
+{
+    /** @var EngineInterface */
+    private $templatingEngine;
+
+    /** @var DashboardStatisticsProviderInterface */
+    private $statisticsProvider;
+
+    /**
+     * @var SalesDataProviderInterface
+     */
+    private $salesDataProvider;
+
+    public function __construct(
+        EngineInterface $templatingEngine,
+        DashboardStatisticsProviderInterface $statisticsProvider,
+        SalesDataProviderInterface $salesDataProvider
+    ) {
+        $this->templatingEngine = $templatingEngine;
+        $this->statisticsProvider = $statisticsProvider;
+        $this->salesDataProvider = $salesDataProvider;
+    }
+
+    public function __invoke(ChannelInterface $channel): Response
+    {
+        return $this->templatingEngine->renderResponse(
+            '@SyliusAdmin/Dashboard/Statistics/_template.html.twig', $this->getRawData($channel,
+            (new \DateTime('first day of next month last year')),
+            (new \DateTime('last day of this month')),
+            'month')
+        );
+    }
+
+    private function getRawData(ChannelInterface $channel, $startDate, $endDate, $interval): array
```

Comment:
I believe it could be the only method in this service (and, of course, public). It seems strange (and too complicated), to have a service that is used both as a controller for rendering templates and that provides data for other services. Let's simplify it - name this service `DashboardDataProvider` with one method `getRawData` returning `array` or some VO and leave the twig- and request/response-related work for the `DashboardController`. WDYT? ⚖️